### ISSUE TITLE
feat(ListKeyManager): Support Signal-based Query

### ISSUE THREAD

**Issue by keatkeat87** [angular/components#28710]
### Feature Description

![image](https://github.com/angular/components/assets/8691013/079f234f-3721-4789-bb17-9e52b5a6e30e)

Signal-based query doesn't return QueryList anymore, it only returns readonly Array.
so ListKeyManager should support it.

and another thing
![image](https://github.com/angular/components/assets/8691013/ba23925c-cc98-42b4-aee9-37ed3f0b0875)
we should have a way to watch Signal like previously watching QueryList.

Workaround:
if array will not changes, we can just cast to not readonly

`const keyManager = new ListKeyManager(this.items() as FocusableOptionDerective[]);`

if we need watch the array, we can access private QueryList from Signal.
```
const signalNode = this.items[SIGNAL] as { _queryList: QueryList<FocusableOptionDerective> };
const keyManager = new ListKeyManager(signalNode._queryList);
```
